cidc-api/services/ingestion.py

```diff
@@ -82,4 +86,70 @@ def validate():
 @ingestion_api.route("/upload", methods=["POST"])
 @requires_auth("ingestion.upload")
 def upload():
+    """Ingest the metadata associated with a completed upload."""
     raise NotImplemented()
+
+
+@ingestion_api.route("/signed-upload-urls", methods=["POST"])
+@requires_auth("ingestion.signed-upload-urls")
+def signed_upload_urls():
+    """
+    Given a request whose body contains a directory name and a list of object names,
+    return a JSON object mapping object names to signed GCS upload URLs for those objects.
+
+    Note: a signed URL gives time-restricted, method-restricted access to one of our GCS
+    storage buckets
+
+    TODO: In the long run, this endpoint *needs* user-level rate-limiting or similar. If we don't keep 
+    track of how recently we've issued signed URLs to a certain user, then that user can
+    keep acquiring signed URLs over and over, effectively circumventing the time-restrictions
+    built into these URLs. For now, though, since only people on the development team are
+    registered in the app, we don't need to worry about this.
+
+    Sample request body:
+    {
+        "directory_name": "my-assay-run-id",
+        "object_names": ["my-fastq-1.fastq.gz", "my-fastq-2.fastq.gz"]
+    }
+
+    Sample response body:
+    {
+        "my-fastq-1.fastq.gz": [a signed URL with PUT permissions],
+        "my-fastq-2.fastq.gz": [a signed URL with PUT permissions]
+    }
+    """
+    # Validate the request body
```

Comment:
Maybe going forward we should use JSON Schema or Cerberus for validations like this?